alicloud/resource_alicloud_polardb_endpoint.go

```diff
@@ -0,0 +1,240 @@
+package alicloud
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/polardb"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func resourceAlicloudPolarDBEndpoint() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudPolarDBEndpointCreate,
+		Read:   resourceAlicloudPolarDBEndpointRead,
+		Update: resourceAlicloudPolarDBEndpointUpdate,
+		Delete: resourceAlicloudPolarDBEndpointDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"db_cluster_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"endpoint_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Custom"}, false),
+				ForceNew:     true,
+			},
+			"db_nodes": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"read_write_mode": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{"ReadWrite", "ReadOnly"}, false),
+				Optional:     true,
+				Default:      "ReadOnly",
+			},
+			"auto_add_new_nodes": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{"Enable", "Disable"}, false),
+				Optional:     true,
+				Default:      "Disable",
+			},
+			"endpoint_config": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudPolarDBEndpointCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	polarDBService := PolarDBService{client}
+	clusterId := d.Get("db_cluster_id").(string)
+	endpointType := d.Get("endpoint_type").(string)
+	request := polardb.CreateCreateDBClusterEndpointRequest()
+	request.RegionId = client.RegionId
+	request.DBClusterId = clusterId
+	request.EndpointType = endpointType
+
+	enpoints, err := polarDBService.DescribePolarDBInstanceNetInfo(clusterId)
+	if err != nil {
+		return WrapError(err)
+	}
+	oldEndpoints := make([]interface{}, 0)
+	for _, value := range enpoints {
+		oldEndpoints = append(oldEndpoints, value.DBEndpointId)
+	}
+	oldEndpointIds := schema.NewSet(schema.HashString, oldEndpoints)
```

Comment:
What does the `oldEndpointIds` do?

Reply:
Openapi doesn't return the newly created endpoint id, before creating, get the id of the existing endpoints to determine the newly created endpoint id.